internal/controller/eventing/integrationtests/validation/integration_test.go

```diff
@@ -0,0 +1,683 @@
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelDebug,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Info`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelInfo,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Warn`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelWarn,
+						},
+					},
+				},
+			},
+		},
+		{
+			name: `validation of spec.logging passes for value = Error`,
+			givenUnstructuredEventing: unstructured.Unstructured{
+				Object: map[string]any{
+					kind:       kindEventing,
+					apiVersion: apiVersionEventing,
+					metadata: map[string]any{
+						name:      test.GetRandK8sName(7),
+						namespace: test.GetRandK8sName(7),
+					},
+					spec: map[string]any{
+						logging: map[string]any{
+							logLevel: logLevelError,
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			// given
+			testEnvironment.EnsureNamespaceCreation(t, tc.givenUnstructuredEventing.GetNamespace())
+
+			// when
+			err := testEnvironment.CreateUnstructuredK8sResource(&tc.givenUnstructuredEventing)
+
+			// then
+			if tc.wantErrMsg == noError {
+				require.NoError(t, err, "Expected error message to be empty but got error instead."+
+					" Check the validation rule of the eventing CR.")
+			} else {
+				require.Error(t, err, fmt.Sprintf("Expected the following error message: \n \" %s \" \n"+
+					" but got no error. Check the validation rules of the eventing CR.", tc.wantErrMsg))
+
+				require.Contains(t, err.Error(), tc.wantErrMsg, "Expected a specific error message"+
+					" but messages do not match. Check the validation rules of the eventing CR.")
+			}
+		})
+	}
+}
+
+// Test_Validate_CreateEventing creates an eventing CR with correct and purposefully incorrect values, and compares
+// the error that was caused by this against a wantErrMsg to test the eventing CR validation rules.
+func Test_Validate_Defaulting(t *testing.T) {
```

Comment:
IMO in every test case we should check the following:
- If there was no given values, we expect the default values to fill these empty values. (Done)
- If there was given values, we expect these values **not** to be overridden by the defaulting logic. (Missing)

Reply:
Wouldn't testing that other values are not overridden by the defaulting logic not be testing our implementation and instead testing if the defaulting markers work correctly?

i.e. not testing our code but testing the OpenAPI v3 schema?

Reply:
That's a valid point.